### ISSUE TITLE
AMD assumes default export when there is none.

### ISSUE THREAD

**Issue by kellyselden** [buxlabs/amd-to-es6#77]
### error

`Error: 'default' is not exported by ...`

### importer

original: https://github.com/emberjs/ember-inflector/blob/master/addon/index.js

babel to AMD:

```js
define('ember-inflector/index', ['exports', 'ember-inflector/lib/system', 'ember-inflector/lib/ext/string'], function (exports, _system) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.defaultRules = exports.singularize = exports.pluralize = undefined;


  _system.Inflector.defaultRules = _system.defaultRules;

  Object.defineProperty(Ember, 'Inflector', {
    get: function get() {
      Ember.deprecate('Ember.Inflector is deprecated. Please explicitly: import Inflector from \'ember-inflector\';', false, {
        id: 'ember-inflector.globals',
        until: '3.0.0'
      });

      return _system.Inflector;
    }
  });

  Object.defineProperty(Ember.String, 'singularize', {
    get: function get() {
      Ember.deprecate('Ember.String.singularize() is deprecated. Please explicitly: import { singularize } from \'ember-inflector\';', false, {
        id: 'ember-inflector.globals',
        until: '3.0.0'
      });

      return _system.singularize;
    }
  });

  Object.defineProperty(Ember.String, 'pluralize', {
    get: function get() {
      Ember.deprecate('Ember.String.pluralize() is deprecated. Please explicitly: import { pluralize } from \'ember-inflector\';', false, {
        id: 'ember-inflector.globals',
        until: '3.0.0'
      });

      return _system.pluralize;
    }
  });

  exports.default = _system.Inflector;
  exports.pluralize = _system.pluralize;
  exports.singularize = _system.singularize;
  exports.defaultRules = _system.defaultRules;
});
```

AMD to ES6:

```js
import _system from 'ember-inflector/lib/system';
import 'ember-inflector/lib/ext/string';
_system.Inflector.defaultRules = _system.defaultRules;
Object.defineProperty(Ember, 'Inflector', {
    get: function get() {
        Ember.deprecate("Ember.Inflector is deprecated. Please explicitly: import Inflector from 'ember-inflector';", false, {
            id: 'ember-inflector.globals',
            until: '3.0.0'
        });
        return _system.Inflector;
    }
});
Object.defineProperty(Ember.String, 'singularize', {
    get: function get() {
        Ember.deprecate("Ember.String.singularize() is deprecated. Please explicitly: import { singularize } from 'ember-inflector';", false, {
            id: 'ember-inflector.globals',
            until: '3.0.0'
        });
        return _system.singularize;
    }
});
Object.defineProperty(Ember.String, 'pluralize', {
    get: function get() {
        Ember.deprecate("Ember.String.pluralize() is deprecated. Please explicitly: import { pluralize } from 'ember-inflector';", false, {
            id: 'ember-inflector.globals',
            until: '3.0.0'
        });
        return _system.pluralize;
    }
});
export default _system.Inflector;
export var pluralize = _system.pluralize;
export var singularize = _system.singularize;
export var defaultRules = _system.defaultRules;
```

### exporter

original: https://github.com/emberjs/ember-inflector/blob/master/addon/lib/system.js

babel to AMD:

```js
define("ember-inflector/lib/system", ["exports", "ember-inflector/lib/system/inflector", "ember-inflector/lib/system/string", "ember-inflector/lib/system/inflections"], function (exports, _inflector, _string, _inflections) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.defaultRules = exports.pluralize = exports.singularize = exports.Inflector = undefined;


  _inflector.default.inflector = new _inflector.default(_inflections.default);

  exports.Inflector = _inflector.default;
  exports.singularize = _string.singularize;
  exports.pluralize = _string.pluralize;
  exports.defaultRules = _inflections.default;
});
```

AMD to ES6:

```js
import _inflector from 'ember-inflector/lib/system/inflector';
import _string from 'ember-inflector/lib/system/string';
import _inflections from 'ember-inflector/lib/system/inflections';
_inflector.default.inflector = new _inflector.default(_inflections.default);
export var Inflector = _inflector.default;
export var singularize = _string.singularize;
export var pluralize = _string.pluralize;
export var defaultRules = _inflections.default;
```

It appears that the babel to AMD step is creating code that assumes a default export from the exporter, when there is none. Then when the AMD to ES6 comes along, it throws on no default export. This seems like either a babel incorrect transpile, or a AMD code flaw, not necessarily something wrong with this library. What do you think?

**Comment by emilos** [comment]
@kellyselden it's hard to say based on the above :) is there a chance that you could provide a repro as a separate repository / branch? that would help a lot

also, just for curiosity, what is the reason for es -> amd -> es transformations?

**Comment by kellyselden** [comment]
It's not ideal. I'm consuming code that is both ES6 and AMD. The fact that some code was converted to AMD is out of my control.

I will work on a reproduction.

**Comment by emilos** [comment]
thanks @kellyselden  :) I reproduced it in the meantime, I didn't notice the issue at first but I think I get it now:

es
```js
var foo = 'foo'
var bar = 'bar'
var baz = 'baz'
export { foo, bar, baz }
```

amd
```
exports.foo = 'foo'
exports.baz = 'baz'
exports.bar = 'bar'
```

es
```js
export var foo = 'foo'
export var bar = 'bar'
export var baz = 'baz'
```

The question is:

Should `exports.foo` be translated to `export { foo: 'foo' }` or `export var foo = 'foo'`?

Helper questions:

1. What output is generated by babel for the following input?

es
```js
export var foo = 'foo'
export var bar = 'bar'
export var baz = 'baz'
```

2. What does the es module spec say about the `export { foo }` syntax?

3. How does such import behave with `--experimental-modules` flag?

I'll try to look for an answer :) but that might take me some time (holidays). You could potentially shorten this time by trying to answer the questions above.

As a temporary workaround you could probably somehow replace:

```js
import system from 'system'
```

to

```js
import * as system from 'system'
```

**Comment by emilos** [comment]
hey @kellyselden I rewrote a large chunk of the code to handle this, it was released as `v0.12.0`. The expected output should be similar to:

```js
var a = pluralize;
var b = singularize;
var c = defaultRules;
var d = Inflector;
export {
  a as pluralize,
  b as singularize,
  c as defaultRules,
  d as default
};
```

which should let you consume the code correctly. Could you please confirm this has helped?

**Comment by kellyselden** [comment]
Will do.
